maven-surefire-plugin/src/site/apt/examples/junit-platform.apt.vm

```diff
@@ -0,0 +1,206 @@
+<dependencies>
+  [...]
+    <dependency>
+      <groupId>org.junit.vintage</groupId>
+      <artifactId>junit-vintage-engine</artifactId>
+      <version>5.2.0</version>
+      <scope>test</scope>
+    </dependency>
+  [...]
+</dependencies>
++---+
+
+* Provider Selection
+
+   If nothing is configured, Surefire detects which JUnit version to use by the following algorithm:
+
++---+
+if the JUnit Platform Engine is present in the project
+    use junit-platform
+if the JUnit version in the project >= 4.7 and the parallel attribute has ANY value
+    use junit47 provider
+if JUnit >= 4.0 is present
+    use junit4 provider
+else
+    use junit3.8.1
++---+
+
+  When using this technique there is no check that the proper test-frameworks are present on your project's
+  classpath. Failing to add the proper test-frameworks will result in a build failure.
+
+* Running Tests in Parallel
+
+  From JUnit Platform does not support running tests in parallel.
+
+* Running a Single Test Class
+
+   The JUnit Platform Provider supports the <<<test>>> JVM system property supported by
+   the Maven Surefire Plugin. For example, to run only test methods in the <<<org.example.MyTest>>> test class
+   you can execute <<<mvn -Dtest=org.example.MyTest test>>> from the command line.
+
+
+* Filtering by Test Class Names
+
+   The Maven Surefire Plugin will scan for test classes whose fully qualified names match
+   the following patterns.
+
+   * <<<**/Test*.java>>>
+
+   * <<<**/*Test.java>>>
+
+   * <<<**/*Tests.java>>>
+
+   * <<<**/*TestCase.java>>>
+
+Moreover, it will exclude all nested classes (including static member classes) by default.
+
+Note, however, that you can override this default behavior by configuring explicit
+`include` and `exclude` rules in your `pom.xml` file. For example, to keep Maven Surefire
+from excluding static member classes, you can override its exclude rules.
+
+Overriding exclude rules of Maven Surefire
+
++---+
+...
+<build>
+    <plugins>
+        ...
+        <plugin>
+            <artifactId>maven-surefire-plugin</artifactId>
+            <version>{surefire-version}</version>
+            <configuration>
+                <excludes>
+                    <exclude/>
+                </excludes>
+            </configuration>
+            ...
+        </plugin>
+    </plugins>
+</build>
+...
++---+
+
+
+* Filtering by Tags
+
+You can filter tests by tags or tag expressions using the following configuration properties.
+
+    * to include <<<tags>>> or <<<tag expressions>>>, use either <<<groups>>> or <<<includeTags>>>.
+
+    * to exclude <<<tags>>> or <<<tag expressions>>>, use either <<<excludedGroups>>> or <<<excludeTags>>>.
+
++---+
+...
+<build>
+    <plugins>
+        ...
+        <plugin>
+            <artifactId>maven-surefire-plugin</artifactId>
+            <version>{surefire-version}</version>
+            <configuration>
+                <properties>
+                    <includeTags>acceptance | !feature-a</includeTags>
```

Comment:
Why we did not utilize Surefire's parameters `groups` and `excludedGroups`?
The same is in TestNG.

Reply:
I think we should indeed do that, cf. https://github.com/junit-team/junit5/issues/1425#issuecomment-390308206

Reply:
I'll take a look.

Reply:
Do you mean with

> Surefire's parameters `groups` and `excludedGroups`

the  TestNG specific contants in `org.apache.maven.surefire.booter.ProviderParameterNames`?

```java
    public static final String TESTNG_EXCLUDEDGROUPS_PROP = "excludegroups";
    public static final String TESTNG_GROUPS_PROP = "groups";
```

If yes, then I'd like to keep the JUnit 5 Platform configuration separate from those. The meaning of entries listed in `groups` and `excludedGroups` may be different.

Reply:
@sormuras 
Let's go to the IRC chat.